Core/Src/main.c

```diff
@@ -143,7 +172,34 @@ void SystemClock_Config(void)
 }
 
 /* USER CODE BEGIN 4 */
+uint16_t ADC_handle(void){
+
+	//Pull Chip select pin low to start connection
+	HAL_GPIO_WritePin(GPIOB, CS_PIN, 0);
```

Comment:
there is a macro you can use to set GPIOs high and low. For example, `GPIO_PIN_RESET` in this case. Just removes the magic number :)